docs/operations.rst

```diff
@@ -14,42 +14,79 @@ Provisioning
-Security
---------
+When upgrading a pg_auto_failover setup, the procedure is different on the
+monitor and on the Postgres nodes:
 
-Connections between monitor and data nodes use *trust* authentication by
-default. This lets accounts used by ``pg_auto_failover`` to connect to nodes
-without needing a password. Default behaviour could be changed using ``--auth``
-parameter when creating monitor or data Node. Any auth method supported by
-PostgreSQL could be used here. Please refer to `PostgreSQL pg_hba documentation`__
-for available options.
+  - on the monitor, the internal pg_auto_failover database schema might have
+    changed and needs to be upgraded to its new definition, porting the
+    existing data over. The pg_auto_failover database contains the
+    registration of every node in the system and their current state.
 
-__ https://www.postgresql.org/docs/current/auth-pg-hba-conf.html
+	It is not possible to trigger a failover during the monitor update.
+	Postgres operations on the Postgres nodes continue normally.
 
-Security for following connections should be considered when setting up
-`.pgpass` file.
+	During the restart of the monitor, the other nodes might have trouble
+	connecting to the monitor. The ``pg_autoctl`` command is designed to
+	retry connecting to the monitor and handle errors gracefully.
 
-  1. health check connection from monitor for `autoctl` user to both `postgres` and `pg_auto_failover` databases.
-  2. connections for `pg_autoctl` command from data nodes to monitor for `autoctl_node` user.
-  3. replication connections from secondary to primary data nodes for `replication` user.
-     Notice that primary and secondary nodes change during failover. Thus this setting
-     should be done on both primary and secondary nodes.
-  4. settings need to be updated after a new node is added.
+  - on the Postgres nodes, the ``pg_autoctl`` command connects to the
+    monitor every once in a while (every second by default), and then calls
+    the ``node_active`` protocol, a stored procedure in the monitor databases.
 
-See `PostgreSQL documentation`__ on setting up `.pgpass` file.
+	The ``pg_autoctl`` also verifies at each connection to the monitor that
+	it's running the expected version of the extension. When that's not the
+	case, the "node-active" sub-process quits, to be restarted with the
+	possibly new version of the ``pg_autoctl`` binary found on-disk.
 
-__ https://www.postgresql.org/docs/current/libpq-pgpass.html
+As a result, here is the standard upgrade plan for pg_auto_failover:
 
+  1. Upgrade the pg_auto_failover package on the all the nodes, monitor
+     included.
 
-Operations
-----------
+	 When using a debian based OS, this looks like the following command for
+	 pg_auto_failover 1.3.1::
+
+	   sudo apt-get remove pg-auto-failover-cli-enterprise-1.0 postgresql-11-auto-failover-enterprise-1.0
+	   sudo apt-get install -q -y pg-auto-failover-cli-enterprise-1.3 postgresql-11-auto-failover-enterprise-1.3
```

Comment:
This upgrade procedure only works for upgrades from 1.4 to future 1.5 right? I think it's good to make that clear somewhere. Pretending like it works to upgrade from 1.0 to 1.3 seems like a recipe for problems. I'm not sure if we need to write down the pre-1.4 upgrade procedure or not, but making clear that this procedure is not the right one for pre-1.4 seems critical.

Reply:
Thinking more about it, I think we should still have a short version of the pre-1.4 procedure. The extra SQL queries we used to have in this document were more confusing than useful, so I re-installed something simpler where you have to run `pg_autoctl enable maintenance` on every standby node.